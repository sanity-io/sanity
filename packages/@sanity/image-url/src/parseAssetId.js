const example = 'image-Tb9Ew8CXIwaY6R1kjMvI0uRR-2000x3000-jpg'

export default function parseAssetId(ref) {
  const [, id, dimensionString, format] = ref.split('-')

  if (!id || !dimensionString || !format) {
    throw new Error(`Malformed asset _ref '${ref}'. Expected an id like "${example}".`)
  }

  const [imgWidthStr, imgHeightStr] = dimensionString.split('x')

  const width = +imgWidthStr
  const height = +imgHeightStr

  const isValidAssetId = isFinite(width) && isFinite(height)
  if (!isValidAssetId) {
    throw new Error(`Malformed asset _ref '${ref}'. Expected an id like "${example}".`)
  }

  return {id, width, height, format}
}
