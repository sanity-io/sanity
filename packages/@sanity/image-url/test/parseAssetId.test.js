import parseAssetId from '../src/parseAssetId'

test('throws on invalid document id', () => {
  expect(() => parseAssetId('moop')).toThrowErrorMatchingSnapshot()
})

test('throws on invalid dimensions', () => {
  expect(() => parseAssetId('image-assetId-mooxmoo-png')).toThrowErrorMatchingSnapshot()
})
