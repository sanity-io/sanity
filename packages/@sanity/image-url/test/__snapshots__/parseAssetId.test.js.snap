// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`throws on invalid dimensions 1`] = `"Malformed asset _ref 'image-assetId-mooxmoo-png'. Expected an id like \\"image-Tb9Ew8CXIwaY6R1kjMvI0uRR-2000x3000-jpg\\"."`;

exports[`throws on invalid document id 1`] = `"Malformed asset _ref 'moop'. Expected an id like \\"image-Tb9Ew8CXIwaY6R1kjMvI0uRR-2000x3000-jpg\\"."`;
